iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -18,6 +18,11 @@ import { escapeRegex } from './util.js'
 import { IXNode } from './ixnode.js';
 import { setDefault } from './util.js';
 import { DocOrderIndex } from './docOrderIndex.js';
+import { MessageBox } from './messagebox.js';
```

Comment:
Arbitrary line:

This is a great solution, thank you! There is a similar issue with spans when the document takes up 3gb in chrome, idk if we could get something similar where we check the memory used and give an error about the doc taking up too much memory, thoughts? (Probably not on this issue, but this solution inspired me)

Reply:
I'm sure we can do something.  I can't see any way to get the DOM memory usage from a browser, so we'd probably have to count the number of elements and use that as a proxy.

I was also planning to extend this to include a limit on the total number of facts.  From what I could tell, the thing that really kills the performance on this document is inserting the wrapper nodes, so we'll hit a similar problem with a large number of facts.

Reply:
Awesome seems good to me, thank you!